### PR TITLE
feat(patient): add input validation

### LINKED ISSUE #1976: Create/Edit Patient inputs are not validated
When creating and editing a patient, only the required `givenName` is validated. The email and tel inputs can be anything and the date of birth can be a day greater than today.

## 🐛 Bug Report

Input fields from Editing/Creating Patients are not validated.

## To Reproduce

Input a text in the telephone field
Input a text that is not an email in the email field
Input a date in 2021 in the date of birth field

The components already give a way to add validation feedback to the user, but it is not used at the time.

### PR DESCRIPTION
**Pull request by LucasBN**
Fixes #1976

**Changes proposed in this pull request:**

- Prefix / Suffix / Given Name / Family Name / Preferred Language Validated to ensure no numerical input
- Added to both edit and create
- Provides feedback to user

<a href="https://gitpod.io/#https://github.com/HospitalRun/hospitalrun-frontend/pull/2011"></a>

### CODE REVIEW

File: src/patients/edit/EditPatient.tsx [hunk 1]
```diff
@@ -104,6 +113,77 @@ const EditPatient = () => {
         }))
       }
     }
+    if (patient.suffix) {
+      if (regexContainsNumber.test(patient.suffix)) {
```
Comment (fox1t) [inline src/patients/edit/EditPatient.tsx:117]: I am not sure that we have to block numbers from being added to suffix and prefix. I think that some cultures can use numbers in this fields. @jackcmeyer what do you think?

Comment (LucasBN) [reply src/patients/edit/EditPatient.tsx:117]: Not sure about prefix but now that you mention it, I think it is possible in a suffix. However usually in Roman numerals...

**Comment by LucasBN** [comment]
> The code generally looks good.
> 
> I've noticed a bug however:
> 
> On any of the fields, once I get an error on a field, I continue to get an error even if I enter a valid value:
> 
> To reproduce
> 
> 1. Enter valid field in given name field
> 2. Enter invalid field in suffix
> 3. Enter invalid field in family name
> 4. Enter valid field in family name
> 5. See that family name is still marked as invalid
> 
> Also, we need to add tests for this new functionality. We should test that each field gets validated.
> 
> I appreciate the contribution! This is great stuff!

Just fixed this bug - simple fix, but there might be a better solution? Will add tests tonight.

File: src/patients/edit/EditPatient.tsx (hunk 1 above)
Comment (fox1t) [reply src/patients/edit/EditPatient.tsx:127]: OK. We can try to implement this and if, in the future, someone needs numeric values we can fix it accordingly. :)

**Comment by tehKapa** [comment]
@LucasBN can you resolve conflicts? Thank you!

**Comment by LucasBN** [comment]
@jackcmeyer @tehKapa The redux changes were quite significant, making resolving the conflicts quite tricky. Could I close this pull request and open another with my code rewritten to work with the redux refactor?

**Comment by jackcmeyer** [comment]
> @jackcmeyer @tehKapa The redux changes were quite significant, making resolving the conflicts quite tricky. Could I close this pull request and open another with my code rewritten to work with the redux refactor?

Sounds food to me! Please let me know if you need/want help. I feel bad that this happened.

**Pull request closed without merging**